connection.c

```diff
@@ -160,12 +199,70 @@ static void handleConnected(const redisAsyncContext *c, int status)
         return;
     }
 
-    /* Call explicit connect callback (even if failed) */
     if (conn->connect_callback) {
         conn->connect_callback(conn);
     }
 }
 
+/* Connection with authentication
+ *
+ * If the connection status is ok, will attempt to authenticate and only then call the connection callback
+ * If connection status is not ok, will immediately transition connection to CONN_CONNECT_ERROR
+ * User callback is called in all cases except when the connection has already been terminated externally
+ */
+
+static void handleConnectedWithAuth(Connection *conn, int status)
+{
+    if (status == REDIS_OK) {
```

Comment:
If we have CONN_TERMINATING, we probably want to avoid an authentication attempt no?